index.js

```diff
@@ -30,8 +30,8 @@ var PATH_REGEXP = new RegExp([
 /**
  * Parse a string for the raw tokens.
  *
- * @param  {String} str
- * @return {Array}
+ * @param  {string} str
+ * @return {!Array}
```

Comment:
By changing the type form `String` to `string`, the compiler by default assumes non-nullability. This could also be achieved by `!String` - which just means a String object that is not null. The Array was changed to be non-nullable as well.